apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/UrlSignerTest.cs

```diff
@@ -179,41 +179,24 @@ private void GetWithCustomerSuppliedEncryptionKeysTest_InitDelayTest()
             var data = _fixture.SmallContent;
             string url = null;
 
-            string key;
-            string hash;
-            using (var rand = RandomNumberGenerator.Create())
-            using (var sha256 = SHA256.Create())
-            {
-                var keyBytes = new byte[32];
-                rand.GetBytes(keyBytes);
-                key = Convert.ToBase64String(keyBytes);
-                hash = Convert.ToBase64String(sha256.ComputeHash(keyBytes));
-            }
-            
-            Func<HttpRequestMessage> createGetRequest = () => new HttpRequestMessage()
+            EncryptionKey key = EncryptionKey.Generate();
+
+            Func<HttpRequestMessage> createGetRequest = () =>
             {
-                Method = HttpMethod.Get,
-                Headers = {
-                    { "x-goog-encryption-algorithm", "AES256" },
-                    { "x-goog-encryption-key", key },
-                    { "x-goog-encryption-key-sha256", hash }
-                }
+                var request = new HttpRequestMessage { Method = HttpMethod.Get };
+                key.ModifyRequest(request);
+                return request;
             };
 
             _fixture.RegisterDelayTest(_duration,
                 beforeDelay: async duration =>
                 {
-                    // TODO: Replace this with calls to the StorageClient if/when it supports customer-supplied encryption keys.
```

Comment:
Adding the EncryptionKey as a parameter to the Sign methods on UrlSigner seems like a good idea. I can add that in a separate PR.

As for my comment about using UploadObjectAsync, I simply mean that everything from here down to line 204 should be replaced by this:
```csharp
await _fixture.Client.UpdateObjectAsync(bucket, name, new UploadObjectOptions { EncryptionKey = key });
```